providers/aws/resources/aws.lr

```diff
@@ -2276,6 +2278,36 @@ private aws.elasticache.cluster @defaults("cacheClusterId region nodeType engine
   transitEncryptionMode string
 }
 
+// Amazon ElastiCache serverless cache
+private aws.elasticache.serverlessCache @defaults("name description status engine engineVersion") {
+  // ARN for the cache
+  arn string
+  // Unique identifier of the serverless cache
+  name string
+  // Description of the serverless cache
+  description string
+  // The name of the cache engine used for this cluster: Memcached or Redis
+  engine string
+  // The version of the cache engine that is used in this cluster
+  engineVersion string
+  // Version number of the engine the serverless cache is compatible with
+  majorEngineVersion string
+  // ID of the Amazon Web Services Key Management Service (KMS) key
+  kmsKeyId string
+  // A list of VPC security groups associated with the cluster
+  securityGroups []aws.ec2.securitygroup
```

Comment:
do we want to fetch those during resource init? we can also make this lazy: `securityGroups()`

Reply:
wrote this as a composable struct so we can potentially use that for more resources